transform/uuid.go

```diff
@@ -0,0 +1,119 @@
+package transform
+
+import (
+	"strings"
+
+	"github.com/buger/jsonparser"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	versionError = SpecError("Please set version 3 || 4 || 5")
+)
+
+// UUID tries to generate a UUID based on spec components
+func UUID(spec *Config, data []byte) ([]byte, error) {
+
+	//grab version
+	version, ok := (*spec.Spec)["version"]
+
+	if !ok {
+		return nil, versionError
+	}
+
+	//grab field where uuid will be placed
+	uuidOutputField, ok := (*spec.Spec)["uuidOutputField"]
+	if !ok {
+		return nil, SpecError("Must provide uuidOutputField")
+	}
+
+	var u uuid.UUID
+	var err error
+
+	switch version {
+	case 4.0:
+		u = uuid.NewV4()
+
+	case 3.0, 5.0:
+
+		names, ok := (*spec.Spec)["names"]
+		if !ok {
+			return nil, SpecError("Must provide names field")
+		}
+
+		nameSpace, ok := (*spec.Spec)["nameSpace"].(string)
+		if !ok {
+			return nil, SpecError("Must provide namesapce, Must be a string")
+		}
+
+		var nameSpaceUUID uuid.UUID
+
+		// swtich on the namespace
+		switch nameSpace {
+		case "DNS":
+			nameSpaceUUID = uuid.NamespaceDNS
+		case "URL":
+			nameSpaceUUID = uuid.NamespaceURL
+		case "OID":
+			nameSpaceUUID = uuid.NamespaceOID
+		case "X500":
+			nameSpaceUUID = uuid.NamespaceX500
+		default:
+			nameSpaceUUID, err = uuid.FromString(nameSpace)
+			if err != nil {
+				return nil, SpecError("nameSpace is not a valid UUID or is not DNS, URL, OID, X500")
+			}
+		}
+
+		nameFields, ok := names.([]interface{})
+		if !ok {
+			return nil, SpecError("Spec is invalid")
+		}
+
+		// loop over the names field
+		for _, field := range nameFields {
+			p, _ := field.(map[string]interface{})["path"].(string)
+
+			name, _, _, err := jsonparser.Get(data, strings.Split(p, ".")...)
```

Comment:
Probably want to use `name, err := getJSONRaw(data, p, false)`, what you have here will not work properly for paths containing array indices.